chsdi/lib/opentransapi/opentransapi.py

```diff
@@ -21,10 +23,22 @@ def get_departures(self, station_id, number_results=5, request_dt_time=False):
         results = self.xml_to_array(api_response_xml)
         return results
 
-    def _format_time(self, str_date_time):
+    def _format_time(self, str_date_time, fmt="%Y-%m-%dT%H:%M:%SZ"):
         from_zone = tz.tzutc()
         to_zone = tz.gettz('Europe/Zurich')
-        date_time = datetime.strptime(str_date_time, '%Y-%m-%dT%H:%M:%SZ')
+
+        try:
+            date_time = datetime.strptime(str_date_time, fmt)
+        except ValueError:
+            # sometimes the timestamp of the OJP 2.0 API's response has 7 digits for the
+            # milliseconds. 6 are expected and only 6 can be handled by Python.
+            # Hence we need to safely truncate everything between the last . and
+            # the +01:00 part of the timestamp, e.g.:
+            # 2024-11-01T15:39:45.5348804+01:00
+            # Use regex to capture and truncate everything between the last '.' and
+            # the first '+' to 6 digits
+            truncated_date_time = re.sub(r'(\.\d{6})\d*(?=\+)', r'\1', str_date_time)
+            date_time = datetime.strptime(truncated_date_time, '%Y-%m-%dT%H:%M:%S.%f%z')
```

Comment:
This function don't need to be a class member, or make it a static method (`@staticmethod`), this way you can reuse it in the unittest instead of copying it.

Reply:
Good idea :+1: